hivemind/dht/__init__.py

```diff
@@ -39,8 +41,9 @@ class DHT(mp.Process):
     * hivemind servers periodically announce their experts via declare_experts (dht_handler.py)
     * trainers find most suitable experts via RemoteMixtureOfExperts (beam_search.py)
 
-    :param initial_peers: one or multiple endpoints pointing to active DHT peers. Similar format to listen_on.
-    :param listen_on: an interface for incoming connections, e.g. "127.0.0.1:*", "0.0.0.0:1234" or "ipv6:[::]:*"
+    :param p2p: instance of hivemind.p2p.P2P that will be used for communication.
```

Comment:
While it is controversial,
i strongly recommend that we
- create P2P inside of DHTNode, not DHT
- forward **kwargs from DHT to DHTNode (hence P2P) instead of using a separate object
- explicitly check for unused kwargs to make errors clearer

The main ratonale is that it is easier from the user perspective:
```python
dht = hivemind.DHT(use_ipfs=True, start=True)
# instead of
dht = hivemind.DHT(p2p=dict(use_ipfs=True), start=True)
```
(2) we can still ensure that there are no errors

Reply:
Things that were already implemented:

- Create P2P inside of DHTNode, not DHT
- Forward **kwargs from DHT to DHTNode

Things I've just implemented after further discussion with @justheuristic:

- Forward **kwargs from DHTNode to hivemind.p2p.P2P (not DHTProtocol)
- Specify all DHTProtocol.create parameters in DHTNode.create explicitly (so, no need to forward **kwargs to DHTProtocol anymore)
- Stop accepting *args and **kwargs in P2P and passing them directly to the daemon (so, unknown arguments will trigger a Python exception instead of a daemon error in CLI)